tests/end_to_end/utils.py

```diff
@@ -0,0 +1,152 @@
+import os
+from typing import List, Tuple
+
+import depthai as dai
+import requests
+from slugs import PARSERS_SLUGS, SLUGS
+
+
+def get_inputs_from_archive(nn_archive: dai.NNArchive) -> List:
+    """Get all inputs from NN archive."""
+    try:
+        inputs = nn_archive.getConfig().model.inputs
+    except AttributeError:
+        print(
+            "This NN archive does not have an input shape. Please use NN archives that have input shapes."
+        )
+        exit(1)
+
+    return inputs
+
+
+def get_input_shape(nn_archive: dai.NNArchive) -> Tuple[int, int]:
+    """Get the input shape of the model from the NN archive."""
+    inputs = get_inputs_from_archive(nn_archive)
+
+    if len(inputs) > 1:
+        raise ValueError(
+            "This model has more than one input. Currently, only models with one input are supported."
+        )
+
+    try:
+        input_shape = inputs[0].shape[2:][::-1]
```

Comment:
This expects the input shape to be NCHW, right? What if it is NHWC?

Reply:
It is fixed in [124e081](https://github.com/luxonis/depthai-nodes/pull/122/commits/124e081c85bab415f408068567b086312b21bf9e).